src/CabalHelper/Compiletime/CompPrograms.hs

```diff
@@ -92,8 +92,11 @@ patchBuildToolProgs SStack progs = do
 createProgSymlink :: FilePath -> FilePath -> IO ()
 createProgSymlink bindir target
   | [exe] <- splitPath target = do
-    Just exe_path <- findExecutable exe
-    createSymbolicLink exe_path (bindir </> takeFileName target)
-  | otherwise = do
-    cwd <- getCurrentDirectory
-    createSymbolicLink (cwd </> target) (bindir </> takeFileName target)
+    mb_exe_path <- findExecutable exe
+    case mb_exe_path of
+      Just exe_path -> createSymbolicLink exe_path (bindir </> takeFileName target)
+      Nothing -> createSymLinkFromCwd
+  | otherwise = createSymLinkFromCwd
+  where createSymLinkFromCwd = do
+          cwd <- getCurrentDirectory
+          createSymbolicLink (cwd </> target) (bindir </> takeFileName target)
```

Comment:
I don't really understand what you're trying to achieve here?

In my mental model `target` should always exist here because we went through `GHC.configure` and consequently `configureProgram`, but I just checked and a program not being found isn't an error for that function. So i suppose what we should do is call `requireProgram` instead of using `getProg` which will throw an error if the program couldn't be found. 

Still how is falling back to guessing the target program is in cwd a good idea here?

Reply:
Oh, it does not what i thought. The strange thing is it fixes the error in azure, i suppose it is by accident. 
I'll try to throw an error instead, following your suggestion.

Reply:
@DanielG i've change the missing patter case to throw an error, with the info available. I hope at least it could help to find the cause